docs/guides/converting_and_upgrading/optax_update_guide.ipynb

```diff
@@ -0,0 +1,430 @@
+          "\n",
+          "You can also refer to the [Optax 101 (quick start)](https://optax.readthedocs.io/en/latest/optax-101.html)."
+        ]
+      },
+      {
+        "cell_type": "markdown",
+        "metadata": {
+          "id": "c47f55048d1e"
+        },
+        "source": [
+          "## Setup and imports\n",
+          "\n",
+          "Install/upgrade Flax in Colab (the Flax package comes with Optax):"
+        ]
+      },
+      {
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {
+          "id": "3d36c6e9da4c"
+        },
+        "outputs": [],
+        "source": [
+          "!pip install --upgrade -q pip jax jaxlib flax"
+        ]
+      },
+      {
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {
+          "id": "f3e05fb841c9"
+        },
+        "outputs": [],
+        "source": [
+          "# Import the necessary libraries\n",
+          "\n",
+          "import flax\n",
+          "import jax\n",
+          "import jax.numpy as jnp\n",
+          "import flax.linen as nn\n",
+          "import optax\n",
+          "\n",
+          "# Note: this is the minimal code required to make below code run.\n",
+          "\n",
+          "batch = {'image': jnp.ones([1, 28, 28, 1]), 'label': jnp.array([0])}\n",
+          "ds_train = [batch]\n",
+          "get_ds_train = lambda: [batch]\n",
+          "model = nn.Dense(1)\n",
+          "variables = model.init(jax.random.key(0), batch['image'])\n",
+          "learning_rate, momentum, weight_decay, grad_clip_norm = .1, .9, 1e-3, 1.\n",
+          "loss = lambda params, batch: jnp.array(0.)\n"
+        ]
+      },
+      {
+        "cell_type": "markdown",
+        "metadata": {
+          "id": "65090e843841"
+        },
+        "source": [
+          "## Replacing ``flax.optim`` with ``optax``"
+        ]
+      },
+      {
+        "cell_type": "code",
+        "execution_count": null,
+        "metadata": {
+          "id": "301328d793ba"
+        },
+        "outputs": [],
+        "source": [
+          "Optax has drop-in replacements for all of Flax's optimizers. Refer to the Optax [Common Optimizers API docs](https://optax.readthedocs.io/en/latest/api.html) for more details.\n",
+          "\n",
+          "The usage is very similar, with some differences that include:\n",
+          "\n",
+          "- Optax (`optax`) does not keep a copy of the `params`, so they need to be passed around separately.\n",
+          "- Flax provides the utility `flax.training.train_state.TrainState` to store the optimizer state, parameters, and other associated data in a single dataclass (not used in the code example below)."
```

Comment:
This section is rendered as a code block instead of a text block in the notebook